test/e2e/framework/resource_creator.go

```
@@ -1,21 +1,165 @@
+	}
+	// get new RESTClient for custom resources
+	crConfig := Global.KubeConfig
+	yamlMap := make(map[interface{}]interface{})
+	err := yaml.Unmarshal(yamlCR, &yamlMap)
+	if err != nil {
+		return nil, err
+	}
+	groupVersion := strings.Split(yamlMap["apiVersion"].(string), "/")
+	crGV := schema.GroupVersion{Group: groupVersion[0], Version: groupVersion[1]}
+	crConfig.GroupVersion = &crGV
+	crConfig.APIPath = "/apis"
+	crConfig.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: scheme.Codecs}
+
+	if crConfig.UserAgent == "" {
+		crConfig.UserAgent = rest.DefaultKubernetesUserAgent()
+	}
+	ctx.CRClient, err = rest.RESTClientFor(crConfig)
+	if err != nil {
+		return nil, err
+	}
+	return ctx.CRClient, nil
+}
+
+func (ctx *TestCtx) createCRFromYAML(yamlFile []byte, resourceName string) error {
+	client, err := ctx.GetCRClient(yamlFile)
+	if err != nil {
+		return err
+	}
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return err
+	}
+	jsonDat, err := y2j.YAMLToJSON(yamlFile)
+	err = client.Post().
+		Namespace(namespace).
+		Resource(resourceName).
+		Body(jsonDat).
+		Do().
+		Error()
+	ctx.AddFinalizerFn(func() error {
+		return client.Delete().
+			Namespace(namespace).
+			Resource(resourceName).
+			Body(metav1.NewDeleteOptions(0)).
+			Do().
+			Error()
+	})
+	return err
+}
+
+func (ctx *TestCtx) createCRDFromYAML(yamlFile []byte) error {
+	decode := extensions_scheme.Codecs.UniversalDeserializer().Decode
+	obj, _, err := decode(yamlFile, nil, nil)
+	if err != nil {
+		return err
+	}
+	switch o := obj.(type) {
+	case *crd.CustomResourceDefinition:
+		_, err = Global.ExtensionsClient.ApiextensionsV1beta1().CustomResourceDefinitions().Create(o)
+		ctx.AddFinalizerFn(func() error {
+			err = Global.ExtensionsClient.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(o.Name, metav1.NewDeleteOptions(0))
+			if err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
+			return nil
+		})
+		return err
```

We should ignore the error if the CRD already exists:
```Go
if apierrors.IsAlreadyExists(err) {
    return nil
}
```